pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -82,6 +82,7 @@ func NewCIDRRangeAllocator(client clientset.Interface, nodeInformer informers.No
 	eventBroadcaster.StartStructuredLogging(0)
 	klog.V(0).Infof("Sending events to api server.")
 	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: client.CoreV1().Events("")})
+	defer eventBroadcaster.Shutdown()
```

Comment:
this has to go after the `<-stopCh`  or in the `Run()` method, otherwise you shutdown the broadcaster when you return the controller initialized